datadog/fwprovider/resource_datadog_tag_pipeline_ruleset_order.go

```
@@ -0,0 +1,256 @@
+		diag.Append(utils.FrameworkErrorDiag(err, fmt.Sprintf("error listing tag pipeline rulesets: %v", httpResponse)))
+		return
+	}
+
+	var existingRulesets []datadogV2.RulesetRespData
+	if respData, ok := resp.GetDataOk(); ok {
+		existingRulesets = *respData
+	}
+
+	// Create a map of existing ruleset IDs for validation
+	existingIDs := make(map[string]bool)
+	for _, ruleset := range existingRulesets {
+		if rulesetID, ok := ruleset.GetIdOk(); ok {
+			existingIDs[*rulesetID] = true
+		}
+	}
+
+	// Validate desired rulesets exist
+	for _, rulesetID := range desiredOrder {
+		if !existingIDs[rulesetID] {
+			diag.AddError("Invalid ruleset ID", fmt.Sprintf("ruleset ID %s does not exist", rulesetID))
+			return
+		}
+	}
+
+	// Create a slice of structs to sort by current position
+	type rulesetWithPosition struct {
+		id       string
+		position int32
+	}
+
+	rulesetPositions := make([]rulesetWithPosition, 0, len(existingRulesets))
+	for _, ruleset := range existingRulesets {
+		if rulesetID, ok := ruleset.GetIdOk(); ok {
+			position := int32(0)
+			if rulesetAttrs, ok := ruleset.GetAttributesOk(); ok {
+				position = rulesetAttrs.GetPosition()
+			}
+			rulesetPositions = append(rulesetPositions, rulesetWithPosition{
+				id:       *rulesetID,
+				position: position,
+			})
+		}
+	}
+
+	// Sort by position to get current order
+	for i := 0; i < len(rulesetPositions); i++ {
+		for j := i + 1; j < len(rulesetPositions); j++ {
+			if rulesetPositions[i].position > rulesetPositions[j].position {
+				rulesetPositions[i], rulesetPositions[j] = rulesetPositions[j], rulesetPositions[i]
+			}
+		}
+	}
+
+	// Create final order: desired order first, then remaining rulesets
+	finalOrder := make([]string, 0, len(rulesetPositions))
+	desiredIDsMap := make(map[string]bool)
+
+	// Add desired rulesets in specified order
+	for _, id := range desiredOrder {
+		finalOrder = append(finalOrder, id)
+		desiredIDsMap[id] = true
+	}
+
+	// Add remaining rulesets in their current order
```

I'd assume we require all rulesets to exist in terraform or they get overwritten. This seems like we'd support reordering by what is referenced here and throwing anything left at the end

---

Initially I was thinking not all rules are managed by terraform when they start using terraform. But we should force the customer to use terraform to manage all their rulesets then start using  datadog_tag_pipeline_rulesets resource. We can just return the error since reordering requires all rulesets ID should be there.